scalar/time.go

```diff
@@ -0,0 +1,71 @@
+package scalar
+
+import (
+	"github.com/apache/arrow/go/v13/arrow"
+)
+
+type Time struct {
+	Int
+	Unit arrow.TimeUnit
+}
+
+func (s *Time) DataType() arrow.DataType {
+	switch {
+	case s.Unit == arrow.Second && s.Int.BitWidth == 32:
+		return arrow.FixedWidthTypes.Time32s
+	case s.Unit == arrow.Millisecond && s.Int.BitWidth == 32:
+		return arrow.FixedWidthTypes.Time32ms
+	case s.Unit == arrow.Nanosecond && s.Int.BitWidth == 64:
+		return arrow.FixedWidthTypes.Time64ns
+	case s.Unit == arrow.Microsecond && s.Int.BitWidth == 64:
+		return arrow.FixedWidthTypes.Time64us
+	default:
+		panic("unknown time unit")
+	}
```

Comment:
`s.Int.BitWidth` may be `0`, need to use wrapper:
```suggestion
	switch width := s.getBitWidth(); width {
	case 64:
		return &arrow.Time64Type{Unit: s.Unit}
	case 32:
		return &arrow.Time32Type{Unit: s.Unit}
	default:
		panic("unsupported bit width " + strconv.Itoa(width))
	}
```

Reply:
We need to make sure that we set proper values, and the code in Apache Arrow already fails on the bad unit/type combination